core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala

```diff
@@ -149,52 +150,47 @@ class AddPartitionsTest extends ZooKeeperTestHarness {
 
     val metaDataForTopic3 = metadata.filter(p => p.topic.equals(topic3)).head
-    val partitionsMetadataForTopic3 = metaDataForTopic3.partitionsMetadata.sortBy(_.partitionId)
-    val partition1DataForTopic3 = partitionsMetadataForTopic3(1)
-    val partition2DataForTopic3 = partitionsMetadataForTopic3(2)
-    val partition3DataForTopic3 = partitionsMetadataForTopic3(3)
-    val partition4DataForTopic3 = partitionsMetadataForTopic3(4)
-    val partition5DataForTopic3 = partitionsMetadataForTopic3(5)
-    val partition6DataForTopic3 = partitionsMetadataForTopic3(6)
-
-    assertEquals(partition1DataForTopic3.replicas.size, 4)
-    assertEquals(partition1DataForTopic3.replicas(0).id, 3)
-    assertEquals(partition1DataForTopic3.replicas(1).id, 2)
-    assertEquals(partition1DataForTopic3.replicas(2).id, 0)
-    assertEquals(partition1DataForTopic3.replicas(3).id, 1)
-
-    assertEquals(partition2DataForTopic3.replicas.size, 4)
-    assertEquals(partition2DataForTopic3.replicas(0).id, 0)
-    assertEquals(partition2DataForTopic3.replicas(1).id, 3)
-    assertEquals(partition2DataForTopic3.replicas(2).id, 1)
-    assertEquals(partition2DataForTopic3.replicas(3).id, 2)
-
-    assertEquals(partition3DataForTopic3.replicas.size, 4)
-    assertEquals(partition3DataForTopic3.replicas(0).id, 1)
-    assertEquals(partition3DataForTopic3.replicas(1).id, 0)
-    assertEquals(partition3DataForTopic3.replicas(2).id, 2)
-    assertEquals(partition3DataForTopic3.replicas(3).id, 3)
-
-    assertEquals(partition4DataForTopic3.replicas.size, 4)
-    assertEquals(partition4DataForTopic3.replicas(0).id, 2)
-    assertEquals(partition4DataForTopic3.replicas(1).id, 3)
-    assertEquals(partition4DataForTopic3.replicas(2).id, 0)
-    assertEquals(partition4DataForTopic3.replicas(3).id, 1)
-
-    assertEquals(partition5DataForTopic3.replicas.size, 4)
-    assertEquals(partition5DataForTopic3.replicas(0).id, 3)
-    assertEquals(partition5DataForTopic3.replicas(1).id, 0)
-    assertEquals(partition5DataForTopic3.replicas(2).id, 1)
-    assertEquals(partition5DataForTopic3.replicas(3).id, 2)
-
-    assertEquals(partition6DataForTopic3.replicas.size, 4)
-    assertEquals(partition6DataForTopic3.replicas(0).id, 0)
-    assertEquals(partition6DataForTopic3.replicas(1).id, 1)
-    assertEquals(partition6DataForTopic3.replicas(2).id, 2)
-    assertEquals(partition6DataForTopic3.replicas(3).id, 3)
+
+    validateLeaderAndReplicas(metaDataForTopic3, 0, 2, Set(2,3,0,1))
+    validateLeaderAndReplicas(metaDataForTopic3, 1, 3, Set(3,2,0,1))
+    validateLeaderAndReplicas(metaDataForTopic3, 2, 0, Set(0,3,1,2))
+    validateLeaderAndReplicas(metaDataForTopic3, 3, 1, Set(1,0,2,3))
+    validateLeaderAndReplicas(metaDataForTopic3, 4, 2, Set(2,3,0,1))
+    validateLeaderAndReplicas(metaDataForTopic3, 5, 3, Set(3,0,1,2))
+    validateLeaderAndReplicas(metaDataForTopic3, 5, 3, Set(3,0,1,2))
+    validateLeaderAndReplicas(metaDataForTopic3, 6, 0, Set(0,1,2,3))
```

Comment:
@ijuma This is actually the order set by the placements logic described in `AdminUtils.assignReplicasToBrokers` [here](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/admin/AdminUtils.scala#L49-L66). I will fix comma spacing.